components/dashboards/org.wso2.carbon.dashboards.core/src/main/java/org/wso2/carbon/dashboards/core/internal/DashboardMetadataProviderImpl.java

```diff
@@ -139,31 +150,97 @@ protected void unsetPermissionManager(PermissionManager permissionManager) {
     }
 
     @Override
-    public Set<DashboardMetadata> getAll() throws DashboardException {
-        return dao.getAll();
+    public Optional<DashboardMetadata> getDashboardByUser(String user, String dashboardUrl, String originComponent)
+            throws DashboardException {
+        // TODO: 11/10/17 validate parameters
+        boolean isAuthorized = originComponent != null ? checkPermissions(user, dashboardUrl, originComponent) :
+                checkPermissions(user, dashboardUrl);
+        if (isAuthorized) {
+            return dao.get(dashboardUrl);
+        } else {
+            throw new DashboardUnauthorizedException("You are not authorized to access the dashboard with ID" +
```

Comment:
Previous error message comment apply here as well. 
`getDashboardByUser` is called by another object (`DashboardRestApi`), not by an user. `"You are ..."` is suitable for an end-user error message, but not for an exception error message.